common/math/integer_test.go

```diff
@@ -114,3 +115,78 @@ func TestMustParseUint64Panic(t *testing.T) {
 	}()
 	MustParseUint64("ggg")
 }
+
+type marshalUnMarshalTest struct {
+	input   interface{}
+	want    interface{}
+	wantErr bool // if true, decoding must fail on any platform
+}
+
+var (
+	marshalHexOrDecimal64Tests = []marshalUnMarshalTest{
+		{input: uint64(0), want: "0x0"},
+		{input: uint64(1), want: "0x1"},
+		{input: uint64(16), want: "0x10"},
+		{input: uint64(255), want: "0xff"},
+		{input: uint64(0xff), want: "0xff"},
+		{input: uint64(0x1122334455667788), want: "0x1122334455667788"},
+	}
+
+	UnMarshalHexOrDecimal64Tests = []marshalUnMarshalTest{
+		// invalid encoding
+		{input: "", wantErr: true},
+		{input: "null", wantErr: true},
+		{input: `"0x"`, wantErr: true},
+		{input: `"0xfffffffffffffffff"`, wantErr: true},
+		{input: `"1ab"`, wantErr: true},
+		{input: `"0xx"`, wantErr: true},
+		{input: `"0x1zz01"`, wantErr: true},
+
+		// valid encoding
+		{input: `""`, want: uint64(0)},
+		{input: `"0"`, want: uint64(0)},
+		{input: `"10"`, want: uint64(10)},
+		{input: `"100"`, want: uint64(100)},
+		{input: `"12344678"`, want: uint64(12344678)},
+		{input: `"1111111111111111"`, want: uint64(1111111111111111)},
+		{input: `"0x0"`, want: uint64(0)},
+		{input: `"0x2"`, want: uint64(0x2)},
+		{input: `"0x2F2"`, want: uint64(0x2f2)},
+		{input: `"0x1122aaff"`, want: uint64(0x1122aaff)},
+		{input: `"0xbbb"`, want: uint64(0xbbb)},
+		{input: `"0xffffffffffffffff"`, want: uint64(0xffffffffffffffff)},
```

Comment:
I think the hex based inputs should be `"\"0x2\""` for example, as the strings are supposed to be JSON, but this doesn't matter with the current implementation of HexOrDecimal64. A follow-up PR to fix this might still be good :)

Reply:
That, or we could call UnmarshalText instead of UnmarshalJSON in the test

Reply:
Agreed, the previous tests covered UnmarshalText so I wanted to test for JSON too, I'll put a followup PR to fix the inputs